lib/logmsg/logmsg.c

```diff
@@ -923,6 +923,21 @@ log_msg_sdata_append_escaped(GString *result, const gchar *sstr, gssize len)
     }
 }
 
+static const gchar *
+first_non_dot_or_digit_char(const gchar *str, gint str_len)
+{
+  const gchar *pos = str;
+
+  while (str_len > 0 && pos && (*pos == '.' || isdigit(*pos)))
+    {
+      ++pos;
+      --str_len;
+    }
+  if (str_len == 0) return NULL;
```

Comment:
using strspn(str, "0123456789.") might be faster (and is a standard part of libc, so no need to open code it.

this is an optional remark.

Reply:
Sure, thanks for the note. The PR needs update either way, so I'll do it.